### PR TITLE
🙉 Enable NACKs behind a feature toggle

### LINKED ISSUE #95: Fix NACKs to more gracefully handle RTP sequence # rollover

### PR DESCRIPTION
**Pull request by danstiner**
Meant to be a minimal set of changes to bring back the NACK. Further refactoring can come later.

Tested on the homelab without/with feature enabled. Enabling this makes a **substantial** improvement in all the simulated packet loss scenarios I've done.

Fixes #95

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,6 +110,7 @@ Configuration is achieved through environment variables.
 | `FTL_SERVICE_CONNECTION` | `DUMMY`: (default) Dummy service connection <br />`GLIMESH`: Glimesh service connection <br />`REST`: REST service connection ([docs](docs/REST_SERVICE.md)) | This configuration value determines which service FTL should plug into for operations such as stream key retrieval. |
 | `FTL_SERVICE_METADATAREPORTINTERVALMS` | Time in milliseconds | Defaults to `4000`, controls how often FTL stream metadata will be reported to the service. |
 | `FTL_MAX_ALLOWED_BITS_PER_SECOND` | Integer bits per second | Defaults to `0` (disabled), FTL connections that exceed the bandwidth specified here will be stopped.<br />**Note that this is a strictly enforced maximum** based on a rolling average; consider providing some buffer size for encoder spikes above the configured average. |
+| `FTL_SERVICE_NACK_LOST_PACKETS` | `0`: Ignore missing packets <br />`1`: Send NACKs | Determines whether to send NACKs for missing packet sequences in the stream. |
```
Comment (haydenmc) [inline README.md:113]: `FTL_SERVICE*` is normally for ServiceConnections - perhaps just `FTL_NACK_LOST_PACKETS` ?

**Comment by danstiner** [comment]
I'm a bit torn how far to go with adding tests to this PR, I think to really make the logic testable it should be pulled into more specialized buffer/nack tracking classes.

Maybe before merging I should add at least one happy path test of relaying packets, and then as we refactor this we can add more tests.

File: src/FtlMediaConnection.cpp [hunk 2]
```diff
@@ -155,17 +162,16 @@ std::set<rtp_sequence_num_t> FtlMediaConnection::insertPacketInSequenceOrder(
...
-        const RtpHeader* cachedRtpHeader = RtpPacket::GetRtpHeader(cachedPacket);
-        const rtp_sequence_num_t cachedSeqNum = ntohs(cachedRtpHeader->SequenceNumber);
+        const RtpPacket& cachedPacket = *it;
+        const auto cachedSeqNum = cachedPacket.ExtendedSequenceNum;
 
-        if (isSequenceNewer(seqNum, cachedSeqNum))
+        if (seqNum > cachedSeqNum)
         {
             // If there are any gaps between the sequence numbers, mark them as missing.
             // cast to uint16_t to account for overflow (sequence numbers wrap from 65535 -> 0)
```
Comment (haydenmc) [inline src/FtlMediaConnection.cpp:171]: This comment is no longer true

File: src/FtlServer.h [hunk 3]
```diff
@@ -66,7 +66,7 @@ class FtlServer : public FtlControlConnectionManager
     /**
      * @brief Starts listening for FTL connections on a new thread.
      */
-    void StartAsync();
+    void StartAsync(bool nackLostPackets);
```
Comment (haydenmc) [inline src/FtlServer.h:69]: Any reason `nackLostPackets` can't be a constructor parameter?

**Review by haydenmc — APPROVED** [review]
Couple nits, then good to merge :)

File: src/FtlServer.h (hunk 3 above)
Comment (danstiner) [reply src/FtlServer.h:69]: Yeah it is weird. The issue is FtlServer is constructed before configuration is loaded in `Init`.
https://github.com/Glimesh/janus-ftl-plugin/blob/master/src/JanusFtl.cpp#L40

We could just move construction of the FtlServer into Init? I think that makes sense, though I'd also then move the control listener construction Init.

**Pull request merged**